instrumentation/src/main/java/io/opentelemetry/android/internal/tools/time/SystemTime.kt

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.android.internal.tools.time
+
+/**
+ * Utility to be able to mock the current system time for testing purposes.
+ *
+ * <p>This class is internal and not for public use. Its APIs are unstable and can change at any
+ * time.
+ */
+internal interface SystemTime {
```

Comment:
True, I still like organizing stuff as much as possible because it'd make it easier for us to find things later if the project becomes too big. But from the user's perspective you're right, just by using this modifier we won't have to be worried about the misuse of these classes (as long as they're used in Kotlin code since you can still access kotlin internal stuff from Java code, though that's probably not something to be too worried about for an Android lib).